File: backend/internal/models/session.go
package models

import "time"

type SessionStatus string
type SessionType string

const (
	SessionConnecting    SessionStatus = "connecting"
	SessionConnected     SessionStatus = "connected"
	SessionDisconnected  SessionStatus = "disconnected"
	SessionError         SessionStatus = "error"
	
	SessionTypeSSH   SessionType = "ssh"
	SessionTypeLocal SessionType = "local"
)

type Session struct {
	ID           string        `json:"id"`
	ConnectionID string        `json:"connection_id"`
	Type         SessionType   `json:"type"`
	Status       SessionStatus `json:"status"`
	ConnectedAt  time.Time     `json:"connected_at,omitempty"`
	Error        string        `json:"error,omitempty"`
	OSType       string        `json:"os_type,omitempty"`
}

File: src/renderer/src/types/session.ts
export type SessionStatus = 'connecting' | 'connected' | 'disconnected' | 'error'
export type SessionType = 'ssh' | 'local'

export interface Session {
  id: string
  connection_id: string
  type: SessionType
  status: SessionStatus
  connected_at?: string
  error?: string
  os_type?: string
}
